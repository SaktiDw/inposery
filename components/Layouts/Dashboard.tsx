import useAuth from "@/helper/hooks/useAuth";
import { Navigation, Sidebar, SidebarItem } from "components";
import { useRouter } from "next/router";
import React, { useEffect } from "react";
import { SWRConfig } from "swr";
import useToggle from "../../helper/hooks/useToggle";

type Props = {
  className?: string;
  children: React.ReactNode;
};
const Dashboard = (props: Props) => {
  const router = useRouter();
  const { toggle, toggler } = useToggle();

  return (
    <main className="bg-slate-100 text-slate-800 dark:bg-slate-900 dark:text-white caret-lime-500 selection:bg-green-700 selection:text-white">
      <Navigation onClick={toggler} />
      <div className="flex h-screen overflow-hidden">
        <Sidebar isOpen={toggle}>
          <SidebarItem
            href="/dashboard"
            text="Dashboard"
            active={router?.pathname == "/dashboard"}
            icon="fi-rr-apps"
          />
          {/* <SidebarItem
            href="#"
            text="Employee"
            active={router?.pathname == "/Employee"}
            icon="fi-rr-users-alt"
          /> */}
          <SidebarItem
            href="/store"
            text="Store"
            active={router?.pathname == "/store"}
            icon="fi-rr-store-alt"
          />
          {/* <SidebarItem
            href="#"
            text="Billings"
            active={router?.pathname == "/Billings"}
            icon="fi-rr-credit-card"
          />
          <SidebarItem
            href="/role"
            text="Roles"
            active={router?.pathname == "/role"}
            icon="fi-rr-id-badge"
          />
          <SidebarItem
            href="/test"
            text="Test"
            active={router?.pathname == "/test"}
            icon="fi-rr-test-tube"
          /> */}
        </Sidebar>
        <div
          className={`w-full h-screen pt-20 pb-10 px-8 overflow-y-auto scroll-smooth ${props.className}`}
        >
          {props.children}
        </div>
      </div>
    </main>
  );
};

export default Dashboard;
