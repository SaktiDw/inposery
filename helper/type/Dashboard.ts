export interface Transaction {
  total: string;
  year: number;
  month: number;
  type: string;
  store_id: number;
}

export interface DashboardResponse {
  id: number;
  name: string;
  user_id: number;
  created_at: Date;
  updated_at: Date;
  total_in: string;
  total_in_yesterday?: any;
  total_in_today: string;
  total_out: string;
  total_out_yesterday?: any;
  total_out_today: string;
  transaction: Transaction[];
}
