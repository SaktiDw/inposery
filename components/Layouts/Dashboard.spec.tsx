import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Dashboard from "./Dashboard";

describe("Dashboard Layout", () => {
  it("should be rendered", () => {
    const wrapper = render(<Dashboard>Dashboard</Dashboard>);
    expect(wrapper).toBeTruthy();
  });

  it("should match with the snapshot", () => {
    const { container } = render(<Dashboard>Dashboard</Dashboard>);
    expect(container).toMatchSnapshot();
  });
});
