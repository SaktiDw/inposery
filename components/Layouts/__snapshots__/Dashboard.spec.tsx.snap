// Vitest Snapshot v1

exports[`Dashboard Layout > should match with the snapshot 1`] = `
<div>
  <main
    class="bg-slate-100 text-slate-800 dark:bg-slate-900 dark:text-white caret-lime-500 selection:bg-green-700 selection:text-white"
  >
    <nav
      class="fixed top-0 left-0 z-50 w-full px-6 py-3 flex items-center bg-slate-100 bg-opacity-50 backdrop-blur dark:bg-slate-900 dark:backdrop-blur-lg dark:bg-opacity-50"
    >
      <button
        class="p-3"
      >
        <i
          class="fi-rr-menu-burger"
        />
      </button>
      <span
        class="font-bold text-2xl"
      >
        In
        <span
          class="text-lime-500"
        >
          POS
        </span>
        ery
      </span>
      <div
        class="flex gap-2 ml-auto"
      >
        <a
          href="/login"
        >
          Login
        </a>
        <a
          href="/"
        >
          Register
        </a>
      </div>
    </nav>
    <div
      class="flex h-screen overflow-hidden"
    >
      <aside
        class="h-screen min-h-screen max-h-screen md:relative pt-16 pb-4 flex flex-col gap-10 overflow-y-auto overflow-x-hidden transition-all ease-in-out duration-200
fixed left-0 bottom-0 top-0 z-20 bg-slate-100 bg-opacity-50 backdrop-blur md:bg-opacity-0
dark:bg-slate-900 dark:backdrop-blur-lg dark:bg-opacity-50
w-[300px]"
      >
        <div
          class=" overflow-y-auto scroll-smooth flex flex-col gap-4 p-4 pb-8 text-slate-500"
        >
          <i
            class="w-10 h-10 rounded-lg flex justify-center items-center transition-all ease-in-out duration-200 shadow group-hover:text-white group-hover:bg-gradient-to-tl from-green-700 to-lime-500  bg-white shadow-xl text-slate-800 dark:bg-slate-800 dark:text-slate-100 fi-rr-apps"
          />
          <span
            class="p-2"
          >
            Dashboard
          </span>
          <i
            class="w-10 h-10 rounded-lg flex justify-center items-center transition-all ease-in-out duration-200 shadow group-hover:text-white group-hover:bg-gradient-to-tl from-green-700 to-lime-500  bg-white shadow-xl text-slate-800 dark:bg-slate-800 dark:text-slate-100 fi-rr-store-alt"
          />
          <span
            class="p-2"
          >
            Store
          </span>
        </div>
        <div
          class="p-4 mx-4 rounded-md shadow-lg bg-gradient-to-tl from-green-700 to-lime-500 mt-auto flex gap-2"
        >
          Pro
        </div>
      </aside>
      <div
        class="w-full h-screen pt-20 pb-10 px-8 overflow-y-auto scroll-smooth undefined"
      >
        Dashboard
      </div>
    </div>
  </main>
</div>
`;
